extra_geom/detectors.py

```diff
@@ -1342,3 +1346,134 @@ def __init__(self, modules, filename='No file'):
         warnings.warn(
             "DSSC_Geometry has been renamed to DSSC_1MGeometry.", stacklevel=2
         )
+
+
+class JUNGFRAUGeometry(DetectorGeometryBase):
+    """Detector layout for flexible Jungfrau arrangements based on the JF-500K
+       module, which is an independent detector unit of 2 x 4 ASIC tiles.
+
+       The ss/fs assignment is the other way around as for AGIPD:
+       y = ss
+       x = fs
+       shape = (ss, fs) = (y, x)
+    """
+    detector_type_name = 'JUNGFRAU'
+    pixel_size = 7.5e-5   # 7.5e-5 metres = 75 micrometer = 0.075 mm
+    frag_ss_pixels = 256  # pixels along slow scan axis within tile
+    frag_fs_pixels = 256  # pixels along fast scan axis within tile
+
+    def __init__(self, modules, filename='No file'):
+        super().__init__(modules, filename)
+        self.expected_data_shape = (len(modules), 512, 1024)
+        self.n_modules = len(modules)
+
+    @classmethod
+    def from_module_positions(cls,offsets=((0,0),), orientations=None,
+                              asic_gap=2, unit=pixel_size):
+        """Generate a Jungfrau geometry object from module positions
+
+           Parameters
+           ----------
+
+           offsets - iterable of length n_modules containing a pixel coordinate
+                     tuple (x,y) for each offset to the global origin
+
+           orientations - iterable of length n_modules containing a unit-vector
+                     tuple (x,y) for each orientation wrt. the axes
+
+           Orientations default to (1,1) for each module if this optional
+           keyword argument is lacking; if not, the number of elements must
+           match the number of modules as per offsets
+
+           We assume that externally defined offsets as per input relate the
+           bottom, beam-left pixel of the module to the global origin, which is
+           bottom, beam-left of the overall assembly.
```

Comment:
The global origin can be anywhere, but our general convention is to use it as the centre of the geometry - or the point where a line from the sample is normal to the detector surface (the PONI, Point Of Normal Incidence in PyFAI terms).

Reply:
Ok. I am aware that the origin is in the center for e.g. AGIPD.  
The way we looked at the 'inspect' scheme (and also assembled data image) for the JUNGFRAU WIP, the origin was bottom-beam-left and so far I kept it like this.  
The difference is that the big detectors have a center by quadrant-design, while for JUNGFRAU this is less obvious. The way the offsets are (currently) defined in the Karabo scene, it seems to suggest strongly a reference to bottom beam-left overall.

In principle one can transform (translate) everything relative to a the nominal pixel center (0,0), which would reflect diffraction geometry better for any radial-symmetric pattern, but for arrangements like 2 x 3 units it will also "split" one module into two coordinate system hemispheres.

Reply:
I'd also consider that the actual origin for a diffraction geometry should be the beam center, which isn't necessarily the nominal detector center (midpoint of assembled pixel image).  
Once the determination of beam center is done - a possible feature that DA can provide in the future, I guess - one can still relate all pixel coordinate to that.

Reply:
Respectively to the PONI ...